### PR TITLE
Make modifications in the flowchart diagram at "Extending Kubernetes" section to tackle incorrect wording

### LINKED ISSUE #43050: Wrong wording in diagram at https://k8s.io/docs/concepts/extend-kubernetes/
In the "Extension point choice flowchart", answering yes to the first questions leads to: Go the "Adding APIs" Section"

There's a grammar error and a punctuation error, but major problem is that there is no such section on that page.

It should read: Go to "API extensions"

### PR DESCRIPTION
**Pull request by ArvindParekh**
**Problem Description:**

The Extension point choice flowchart at [link](https://kubernetes.io/docs/concepts/extend-kubernetes/#extension-flowchart) contains a grammatical error and mentions a section which is not present in the docs (i.e. the "Adding APIs" section) 

**Changes Made:**

- Re-created the flowchart to include the correct section (i.e. the "API Extensions" section)
- Increased the font size to make it more readable.

Fixes #43050 

**Screenshots:**

Before:
image

After:
image

### CODE REVIEW

**Review by Gauravpadam — APPROVED** [review]
1. I like the idea of white text in the flowchart, It's more readable now
2. I like how you've eliminated the negative space from the image, The image looks better now

Overall, lgtm

File: content/en/docs/concepts/extend-kubernetes/flowchart.png
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/flowchart.png:1]: - Please don't use a lower resolution chart. You can keep the same raster height and width, or you can switch it to SVG
- If changing colors, please consider using Kubernetes colors

Comment (ArvindParekh) [reply content/en/docs/concepts/extend-kubernetes/flowchart.png:1]: Thanks, Tim.

- Oh yes, it slipped my mind, I'll go with the SVG
- I thought I only had to make the textual correction in this one and then a second PR for the color changes, based on our discussion in #43050

But I guess it'll be easier to do both in the same PR, this is the updated diagram with Kubernetes colors (Kubernetes blue-> #326CE6, Kubernetes orange-> #D95E08, Kubernetes green -> #4CAF50 and Kubernetes gray -> #333333) that I'll be using in my next commit. 
Please have a look and let me know in case of any changes:

kubernetes drawio (1)

Comment (sftim) [reply content/en/docs/concepts/extend-kubernetes/flowchart.png:1]: You can leave the colors as-is. Right now this PR _is_ making color changes - to some of the text. I'm fine with either **no styling changes at all**, or recoloring to match our house styles better.

- If recoloring, I would make the rhombus shapes white text on blue, with no stroke color, and use gray (again, no stroke color) for the YES / NO ellipses.
- To help with accessibility, consider making the YES and NO text larger and also bold

Comment (ArvindParekh) [reply content/en/docs/concepts/extend-kubernetes/flowchart.png:1]: Got it! 
- I've made the suggested changes and used #326CE6 for blue and well, experimented with different shades - #333333, #454545 and #7E7E7E, for gray. 
  [Some googling suggested me #333333 as the widely used gray in and around Kubernetes, but it closely resembles black, so I thought of comparing it with *slightly lighter* shades; which reminds me, maybe a dedicated section in the style guide of the docs mentioning the Kubernetes color codes?]

- Also increased the font size from 16 to 18 for better visibility and made the text bold for YES and NO ellipses.

Here's how the website looks using the following shades of gray:

1. #333333
image
#333333-background drawio

2. #454545
image
#454545-background drawio

3. #7E7E7E
image
#7E7E7E-background drawio

(Excuse me, if the website screenshots look purple-ish, there might be some issue with my computer)

I personally like the 2nd and 3rd ones, but please let me know which one you'd like me to go with or if there are any changes!

Comment (sftim) [reply content/en/docs/concepts/extend-kubernetes/flowchart.png:1]: Let's go with `#454545` gray, it's better for accessibility.

Comment (ArvindParekh) [reply content/en/docs/concepts/extend-kubernetes/flowchart.png:1]: Alright, great!

File: content/en/docs/concepts/extend-kubernetes/_index.md [hunk 1]
```diff
@@ -3,10 +3,10 @@ title: Extending Kubernetes
...
 description: Different ways to change the behavior of your Kubernetes cluster.
 reviewers:
-- erictune
-- lavalamp
-- cheftako
-- chenopis
+  - erictune
+  - lavalamp
+  - cheftako
+  - chenopis
```
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/_index.md:9]: If you can leave these lines unchanged, even better.

File: content/en/docs/concepts/extend-kubernetes/_index.md [hunk 2]
```diff
@@ -37,29 +37,29 @@ This document is primarily about _extensions_.
...
-* [`kube-apiserver`](/docs/reference/command-line-tools-reference/kube-apiserver/)
-* [`kube-controller-manager`](/docs/reference/command-line-tools-reference/kube-controller-manager/)
-* [`kube-scheduler`](/docs/reference/command-line-tools-reference/kube-scheduler/)
-* [`kubelet`](/docs/reference/command-line-tools-reference/kubelet/)
-* [`kube-proxy`](/docs/reference/command-line-tools-reference/kube-proxy/)
+- [`kube-apiserver`](/docs/reference/command-line-tools-reference/kube-apiserver/)
+- [`kube-controller-manager`](/docs/reference/command-line-tools-reference/kube-controller-manager/)
+- [`kube-scheduler`](/docs/reference/command-line-tools-reference/kube-scheduler/)
+- [`kubelet`](/docs/reference/command-line-tools-reference/kubelet/)
+- [`kube-proxy`](/docs/reference/command-line-tools-reference/kube-proxy/)
```
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/_index.md:47]: If you can leave these lines unchanged, even better.

File: content/en/docs/concepts/extend-kubernetes/_index.md [hunk 3]
```diff
@@ -37,29 +37,29 @@ This document is primarily about _extensions_.
...
-Built-in *policy APIs*, such as [ResourceQuota](/docs/concepts/policy/resource-quotas/),
+Built-in _policy APIs_, such as [ResourceQuota](/docs/concepts/policy/resource-quotas/),
 [NetworkPolicy](/docs/concepts/services-networking/network-policies/) and Role-based Access Control
 ([RBAC](/docs/reference/access-authn-authz/rbac/)), are built-in Kubernetes APIs that provide declaratively configured policy settings.
 APIs are typically usable even with hosted Kubernetes services and with managed Kubernetes installations.
 The built-in policy APIs follow the same conventions as other Kubernetes resources such as Pods.
 When you use a policy APIs that is [stable](/docs/reference/using-api/#api-versioning), you benefit from a
 [defined support policy](/docs/reference/using-api/deprecation-policy/) like other Kubernetes APIs.
-For these reasons, policy APIs are recommended over *configuration files* and *command arguments* where suitable.
+For these reasons, policy APIs are recommended over _configuration files_ and _command arguments_ where suitable.
```
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/_index.md:62]: If you can leave these lines unchanged, even better.

File: content/en/docs/concepts/extend-kubernetes/_index.md
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/_index.md:1]: The image changes are good, but: please watch out for your text editor making unrelated changes to the Markdown source.

File: content/en/docs/concepts/extend-kubernetes/flowchart.svg
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/flowchart.svg:1]: Nice!

File: content/en/docs/concepts/extend-kubernetes/_index.md
Comment (ArvindParekh) [reply content/en/docs/concepts/extend-kubernetes/_index.md:1]: Thanks, Tim. 
I agree, there's some issue that's **automatically** making these unwanted changes in the markdown file and I didn't even realise these changes were made until after I pushed my branch, and I can't really figure out what it is. I've tried editing the file in Vim as well as Nano, but the issue's still there. I've explained it in depth in the slack's `#sig-docs` channel, to try and get some help. 
I'd appreciate any help with this matter!

Comment (dipesh-rawat) [reply content/en/docs/concepts/extend-kubernetes/_index.md:1]: @ArvindParekh In your video shared on sig-docs, just noticed that you've installed the "[Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)" VS Code extension in the lower right corner. It's possible that it's configured to automatically format on save, which could be leading to unintended alterations. You could try uninstalling or disabling it to see if that resolves the issue.

Comment (ArvindParekh) [reply content/en/docs/concepts/extend-kubernetes/_index.md:1]: @dipesh-rawat Thank you for the suggestion, it worked! Prettier was causing issues with the markdown file.
I've updated the file and pushed the changes.

**Comment by tengqm** [comment]
/lgtm
Changing png to svg is a good move.

**Review by sftim — COMMENTED** [review]
/approve
